src/commands/PopulateDatabase.php

```diff
@@ -94,6 +94,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $output->writeln('Dropping current database and loading structure...');
         $this->dropAndInitDb();
 
+        // wait for sql server to settle.
+        // Not sure why it is necessary but without it the config settings from the yaml file are not inserted
+        sleep(45);
```

Comment:
look at this, I think it helped: 
![2023-09-15-113646_1569x262_scrot](https://github.com/elabftw/elabftw/assets/3043706/4b49c158-d3d7-41bc-8501-2e177f324ac6)

this is a commit from nico-cat branch

Reply:
This does not do the trick.
I tried to dump Config::readAll() to see whats happening but the result is empty when called from within PopulateDatabase without the sleep command.

~~~diff
     public function patch(Action $action, array $params): array
-        return $this->readAll();
+        $res = $this->readAll();
+        var_dump($res);
+        return $res;
     }
~~~

However, there seems to be a difference between the very first run (`yarn unit`) with an empty database (new docker container) and a second run when the database was populated before (reuse of the docker container).

Reply:
I tried to find a better solution but all my attempts failed.
So [this commit](https://github.com/elabftw/elabftw/pull/4595/commits/9423588c090c4ac5580acadeb84e4235d83fd060) it is.

If you @NicolasCARPi know a better solution and/or understand the underlying problem better, I'm all ears.
Maybe something with flush tables or locks?